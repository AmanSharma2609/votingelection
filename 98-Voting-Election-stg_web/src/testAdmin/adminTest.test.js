import { shallow, configure, render, mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from 'react-dom';
import { BrowserRouter  } from 'react-router-dom';



import LoginPage from '../components/adminModule/loginPage'

configure({ adapter: new Adapter() });


describe('Login component', () => {
  it('Login: renders correctly', () => {
    const wrapper = shallow(
      <LoginPage />
      )
    expect(wrapper.find(<LoginPage />)).toMatchSnapshot();
  });
  // it("should create an entry in component state with the event value", () => {
  //   // given
  //   const component = mount(<BrowserRouter>
  //     <LoginPage />
  //     </BrowserRouter>);
  //   const form = component.find('TextField');
  //   // when
  //   form.props().onChange({target: {
  //     name: 'myName',
  //     value: 'myValue'
  //   }});
  //   // then
  //   expect(component.state('admin_email')).toEqual('myValue');
  // });
})
  // it('should toggle the state.disabled property when clicking on first button', () => {
  //   const wrapper = shallow(<LoginPage />);
  //   const firstIconButton = wrapper.find('IconButton');
  //   expect(wrapper.state()).toEqual(true);

  //   expect(wrapper.state()).toEqual(false);
  
  // })
//   it('not has ActivityIndicator when state loading false', () => {
//     const wrapper = shallow(<LoginPage />);
  
//     c.instance().handleClickShowPassword().then(() => {
//       expect(c.instance().state.showPassword).toEqual(false);
//     });
//     expect(c.instance().state.isLoading).toEqual(true);
// })
// const c = shallow(<LoginPage />);
// c.instance().handleClickShowPassword().then(() => {
//   expect(c.instance().state.showPassword).toEqual(false);
// });
// expect(c.instance().state.showPassword).toEqual(true);
// });
 