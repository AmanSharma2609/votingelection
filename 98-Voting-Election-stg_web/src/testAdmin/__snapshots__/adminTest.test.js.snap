// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Login: renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <WithStyles(withRouter(LoginPage)) />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "buttonLogin": "withRouter-LoginPage--buttonLogin-5",
          "container": "withRouter-LoginPage--container-1",
          "resize": "withRouter-LoginPage--resize-4",
          "textFieldInput1": "withRouter-LoginPage--textFieldInput1-2",
          "textFieldInput2": "withRouter-LoginPage--textFieldInput2-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "classes": Object {
            "buttonLogin": "withRouter-LoginPage--buttonLogin-5",
            "container": "withRouter-LoginPage--container-1",
            "resize": "withRouter-LoginPage--resize-4",
            "textFieldInput1": "withRouter-LoginPage--textFieldInput1-2",
            "textFieldInput2": "withRouter-LoginPage--textFieldInput2-3",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "buttonLogin": "withRouter-LoginPage--buttonLogin-5",
          "container": "withRouter-LoginPage--container-1",
          "resize": "withRouter-LoginPage--resize-4",
          "textFieldInput1": "withRouter-LoginPage--textFieldInput1-2",
          "textFieldInput2": "withRouter-LoginPage--textFieldInput2-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
}
`;
